src/TaskLineRenderer.ts

```diff
@@ -31,6 +31,55 @@ export const LayoutClasses: { [c in TaskLayoutComponent]: string } = {
     blockLink: '',
 };
 
+type AttributeValueCalculator = (component: TaskLayoutComponent, task: Task) => string | null;
+
+export class AppleSauce {
```

Comment:
I didn't check if this was a draft PR or not. But just in case it's not obvious, AppleSauce is fine for during the refactoring process but not for any merged code...

Reply:
How about:

`AppleSauce` -> `FieldLayoutDetail`
`appleSauceDictionary` -> `FieldLayouts`

In general it's not useful to add types like `Dictionary` to names - it reveals implementation details - and if the type changes, you have to spend time renaming uses...

Reply:
Awesome. How do you feel about extracting `FieldLayoutDetail`, `FieldLayouts`and `FieldLayoutDetail`'s tests to separate files `FieldLayoutDetail.ts` & `FieldLayoutDetail.test.ts`?

Reply:
I don't know, but definitely not for this PR - it's much easier to understand this when you see the old and new code side-by-side in the diffs...

Reply:
Ok